packages/core/react/src/getClusterFromEndpoint.ts

```diff
@@ -0,0 +1,12 @@
+import type { Connection, Cluster } from '@solana/web3.js';
+
+export default function getClusterFromEndpoint(connection?: Connection): Cluster {
+    if (!connection) return 'mainnet-beta';
+    // HACK: remove casts when https://github.com/solana-labs/solana/pull/28435 lands
+    if ((connection as any).cluster) return (connection as any).cluster;
+    if (/devnet/i.test(connection.rpcEndpoint)) return 'devnet';
+    if (/testnet/i.test(connection.rpcEndpoint)) return 'testnet';
+    // HACK: remove casts when https://github.com/solana-labs/solana/pull/28435 lands
+    if (/localhost/i.test(connection.rpcEndpoint)) return 'localnet' as Cluster;
```

Comment:
This is an error from the perspective of the Mobile Wallet Adapter protocol. `localnet` is not a cluster for which you can seek an authorization, so this will always fail.

Reply:
Right, that was a question I had. What should happen if a dapp does this?

Reply:
It'll throw `ERROR_CLUSTER_NOT_SUPPORTED` from the protocol implementation. https://solana-mobile.github.io/mobile-wallet-adapter/spec/spec.html#errors

Reply:
We could say "you said you want to use localhost, don't even add MWA by default since it won't work"

Reply:
That makes sense. Short circuit that in `WalletAdapter` to not add the MWA in the first place. Land this and I can add that as a follow up!

Otherwise the dApp should just die. Until such time as you can run a validator on your phone, there's no sense in authorizing for `localnet` in a mobile wallet.

Reply:
Yeah, makes sense. Will do, and make the other changes. Open question is whether the PR into web3.js is even a good idea, so maybe worth deciding that first.

Reply:
Replied on that PR!